triton-inference-client.spec

```diff
@@ -17,57 +15,20 @@ Requires: protobuf grpc cuda abseil-cpp re2
 
 # locations of CMakeLists.txt
 PROJ_DIR=../%{n}-%{realversion}/src/c++
-CML_CPP=${PROJ_DIR}/CMakeLists.txt
-CML_LIB=${PROJ_DIR}/library/CMakeLists.txt
-
-# remove rapidjson dependence
-sed -i '/RapidJSON CONFIG REQUIRED/,+13d;' ${CML_LIB}
-sed -i '/triton-common-json/d' ${CML_LIB}
-# core repo not needed for grpc-client-only install
-sed -i '/FetchContent_MakeAvailable(repo-core)/d' ${CML_CPP}
-# remove attempts to install external libs
-sed -i '\~/../../_deps/repo-third-party-build/~d' ${CML_LIB}
-sed -i '\~/../../third-party/~d' ${CML_LIB}
-# keep typeinfo in .so by removing ldscript from properties
-sed -i '/set_target_properties/,+5d' ${CML_LIB}
-# change flag due to bug in gcc10 https://gcc.gnu.org/bugzilla/show_bug.cgi?id=95148
-if [[ `gcc --version | head -1 | cut -d' ' -f3 | cut -d. -f1,2,3 | tr -d .` -gt 1000 ]] ; then 
-    sed -i -e "s|Werror|Wtype-limits|g" ${CML_LIB}
-fi
-
-# these files were extracted from:
-# https://github.com/triton-inference-server/server/blob/v2.11.0/src/core/model_config.h
-# https://github.com/triton-inference-server/server/blob/v2.11.0/src/core/model_config.cc
-cp %{_sourcedir}/model_config.h  ${PROJ_DIR}/library/
-cp %{_sourcedir}/model_config.cc ${PROJ_DIR}/library/
-
-# add custom header to cmake build
-sed -i 's/grpc_client.cc common.cc/& model_config.cc/' ${CML_LIB}
-sed -i 's/grpc_client.h common.h/& model_config.h/' ${CML_LIB}
-sed -i '\~${CMAKE_CURRENT_SOURCE_DIR}/common.h~a ${CMAKE_CURRENT_SOURCE_DIR}/model_config.h' ${CML_LIB}
 
 rm -rf ../build
 mkdir ../build
 cd ../build
 
-common_tag_2_11_0=249232758855cc764c78a12964c2a5c09c388d87
-mkdir repo-common && pushd repo-common && curl -k -L https://github.com/%{github_user}/common/archive/${common_tag_2_11_0}.tar.gz | tar -xz --strip=1 && popd
+common_tag=d5c561841e9bd0818c40e5153bdb88e98725ee79
+mkdir repo-common && pushd repo-common && curl -k -L https://github.com/%{github_user}/common/archive/${common_tag}.tar.gz | tar -xz --strip=1 && popd
```

Comment:
@kpedro88 , I never noticed that we are downloading the sources here. Why not just make this an extra source 
```
%define common_tag d5c561841e9bd0818c40e5153bdb88e98725ee79
Source1: https://github.com/%{github_user}/common/archive/%{common_tag}.tar.gz
```

this way it will be cached in cmsrep for future use.

Reply:
thanks for the suggestion, implemented